File: src/algorithms/arrays/cyclic-sort/cyclic-sort/sources/cyclic-sort_test.rs
include!("cyclic-sort.rs");

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_basic_unsorted() {
        assert_eq!(cyclic_sort(&[3, 5, 2, 1, 4]), vec![1, 2, 3, 4, 5]);
    }

    #[test]
    fn test_already_sorted() {
        assert_eq!(cyclic_sort(&[1, 2, 3, 4]), vec![1, 2, 3, 4]);
    }

    #[test]
    fn test_reverse_sorted() {
        assert_eq!(cyclic_sort(&[5, 4, 3, 2, 1]), vec![1, 2, 3, 4, 5]);
    }

    #[test]
    fn test_single_element() {
        assert_eq!(cyclic_sort(&[1]), vec![1]);
    }

    #[test]
    fn test_empty_array() {
        assert_eq!(cyclic_sort(&[]), vec![]);
    }

    #[test]
    fn test_two_elements_swapped() {
        assert_eq!(cyclic_sort(&[2, 1]), vec![1, 2]);
    }

    #[test]
    fn test_default_input() {
        assert_eq!(cyclic_sort(&[3, 5, 2, 1, 4, 6]), vec![1, 2, 3, 4, 5, 6]);
    }

    #[test]
    fn test_longer_array() {
        assert_eq!(
            cyclic_sort(&[8, 3, 6, 1, 5, 9, 2, 7, 4, 10]),
            vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
        );
    }
}

File: src/algorithms/arrays/cyclic-sort/cyclic-sort/cyclic-sort_test.py
import importlib
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "sources"))

cyclic_sort_module = importlib.import_module("cyclic-sort")
cyclic_sort = cyclic_sort_module.cyclic_sort


def test_basic_unsorted():
    assert cyclic_sort([3, 5, 2, 1, 4]) == [1, 2, 3, 4, 5]


def test_already_sorted():
    assert cyclic_sort([1, 2, 3, 4]) == [1, 2, 3, 4]


def test_reverse_sorted():
    assert cyclic_sort([5, 4, 3, 2, 1]) == [1, 2, 3, 4, 5]


def test_single_element():
    assert cyclic_sort([1]) == [1]


def test_empty_array():
    assert cyclic_sort([]) == []


def test_two_elements_swapped():
    assert cyclic_sort([2, 1]) == [1, 2]


def test_default_input():
    assert cyclic_sort([3, 5, 2, 1, 4, 6]) == [1, 2, 3, 4, 5, 6]


def test_does_not_mutate():
    original = [3, 5, 2, 1, 4]
    cyclic_sort(original)
    assert original == [3, 5, 2, 1, 4]


def test_longer_array():
    result = cyclic_sort([8, 3, 6, 1, 5, 9, 2, 7, 4, 10])
    assert result == [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


if __name__ == "__main__":
    test_basic_unsorted()
    test_already_sorted()
    test_reverse_sorted()
    test_single_element()
    test_empty_array()
    test_two_elements_swapped()
    test_default_input()
    test_does_not_mutate()
    test_longer_array()
    print("All tests passed!")
